main/main.go

```diff
@@ -133,11 +135,19 @@ func main() {
 	if logLevel <= logging.Debug {
 		networkLogLevel = logLevel
 	}
+	var trackedSubnets set.Set[ids.ID]
+	for _, sourceBlockchain := range cfg.SourceBlockchains {
+		trackedSubnets.Add(sourceBlockchain.GetSubnetID())
+	}
+
 	network, err := peers.NewNetwork(
 		networkLogLevel,
 		prometheus.DefaultRegisterer,
+		trackedSubnets,
 		&cfg,
 	)
+	network.InitializeConnectionsAndCheckStake(&cfg)
```

Comment:
does the config nee to be passed by reference

Reply:
Not strictly no, since it gets used as interface but from what I've seen we normally pass configs as references and I've done that in the past as well. Not a big deal in general but in applications where configs get passed around a lot doing this can save on allocs and garbage collection.

Reply:
can we remove the pass by ref here, want to avoid it being unintentionally altered

Reply:
FWIW configs are being passed by reference everywhere in this codebase. As mentioned in previous comment we can clean this up but I'd like to punt on this until later

Reply:
same as above, good on punting but can you create an issue to track

Reply:
Updated: https://github.com/ava-labs/awm-relayer/issues/409